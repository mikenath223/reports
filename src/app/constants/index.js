export const pieColors = ["#A259FF", "#F24E1E", "#FFC107", "#6497B1"];
