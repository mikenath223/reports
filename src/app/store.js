import { configureStore } from "@reduxjs/toolkit";
import logger from "redux-logger";
import projectsReducer from "./slices/projectsSlice";
import formReducer from "./slices/formSlice";
import userReducer from "./slices/userSlice";
import gatewayReducer from "./slices/gatewaySlice";
import reportsReducer from "./slices/reportsSlice";

const middlewares = [];
if (process.env.NODE_ENV === `development`) {
  middlewares.push(logger);
}

export const store = configureStore({
  reducer: {
    projects: projectsReducer,
    form: formReducer,
    users: userReducer,
    gateways: gatewayReducer,
    reports: reportsReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(middlewares),
});
