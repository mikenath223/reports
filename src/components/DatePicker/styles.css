.ReactModal__Content {
  width: max-content;
  height: max-content;
  position: unset !important;
}

.ReactModal__Overlay {
  display: flex;
  justify-content: center;
  align-items: center;
}

.date-text {
  background-color: #1bc5bd;
  color: #fff;
  border-radius: 5px;
  padding: 8px 10px;
  border: none;
  cursor: pointer;
}

.task-img {
  padding-top: 3px;
  margin-left: 5px;
}
