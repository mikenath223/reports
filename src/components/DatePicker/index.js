import Modal from "react-modal";
import Calendar from "react-calendar";
import { ReactComponent as Task } from "app/images/calendar.svg";
import PropTypes from "prop-types";
import "react-calendar/dist/Calendar.css";
import "./styles.css";

export default function DatePicker({
  text,
  date,
  dateText,
  handleChange,
  isOpen,
  setOpen,
  minDate,
}) {
  return (
    <div className="datewrap">
      <button className="date-text" onClick={() => setOpen(true)}>
        <p>
          {text}
          <Task className="task-img" />
        </p>
        <p>{dateText}</p>
      </button>
      <Modal
        isOpen={isOpen}
        onRequestClose={() => setOpen(false)}
        contentLabel="Calendar modal"
      >
        <Calendar
          onChange={(date) => {
            handleChange(new Date(date).toDateString());
            setOpen(false);
          }}
          value={new Date(date)}
          minDate={minDate}
        />
      </Modal>
    </div>
  );
}

DatePicker.propTypes = {
  text: PropTypes.string,
  date: PropTypes.string,
  dateText: PropTypes.string,
  handleChange: PropTypes.func,
  isOpen: PropTypes.bool,
  setOpen: PropTypes.func,
  minDate: PropTypes.instanceOf(Date),
};
