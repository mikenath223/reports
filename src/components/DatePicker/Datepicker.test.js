import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "app/store";
import DatePicker from ".";

describe("DatePicker", function () {
  it("should match snapshot", () => {
    const { container } = render(
      <Provider store={store}>
        <DatePicker />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
