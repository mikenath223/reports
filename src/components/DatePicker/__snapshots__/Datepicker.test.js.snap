// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker should match snapshot 1`] = `
<div>
  <div
    class="datewrap"
  >
    <button
      class="date-text"
    >
      <p>
        <svg
          class="task-img"
        >
          calendar.svg
        </svg>
      </p>
      <p />
    </button>
  </div>
</div>
`;
